spec/01-unit/006-schema_validation_spec.lua

```diff
@@ -292,6 +292,22 @@ describe("Schemas", function()
         assert.falsy(err)
         assert.same({"hello", "world"}, values.array)
       end)
+
+      it("should handle escaped commas in comma-separated arrays", function()
+        local s = {
+          fields = {
+            array = {type = "array"}
+          }
+        }
+
+        -- It should also strip the resulting strings
+        local values = { array = "hello\\, world,goodbye world\\,,/hello/\\d{1\\,3}" }
+
+        local valid, err = validate_entity(values, s)
+        assert.True(valid)
+        assert.falsy(err)
+        assert.same({"hello, world", "goodbye world,", "/hello/\\d{1,3}"}, values.array)
```

Comment:
I think strip meant the same as trim there. Anyway, I removed the trimming to make sure the unit test tests a single thing, and otherwise applied all your suggested changes! Thanks for the detailed review!